docs/linters.md

```diff
@@ -437,6 +438,37 @@ linterConfig:
         message: prefer 'colour' over 'color' when referring to colours in field names
 ```
 
+## NumericBounds
+
+The `numericbounds` linter checks that numeric fields (`int32`, `int64`, `float32`, `float64`) have appropriate bounds validation markers.
+
+According to Kubernetes API conventions, numeric fields should have bounds checking to prevent values that are too small, negative (when not intended), or too large.
+
+This linter ensures that:
+- Numeric fields have both `+kubebuilder:validation:Minimum` and `+kubebuilder:validation:Maximum` markers
+- K8s declarative validation markers (`+k8s:minimum` and `+k8s:maximum`) are also supported
+- Bounds values are within the type's valid range (e.g., int32 bounds must fit in int32)
+- `int64` fields with bounds outside the JavaScript safe integer range (±2^53-1) are flagged
+
+**Note:** While `+k8s:minimum` is documented in the official Kubernetes declarative validation spec, `+k8s:maximum` is not yet officially documented but is supported by this linter for forward compatibility and consistency.
+
+### Configuration
+
+This linter is **not enabled by default** as it is primarily focused on CRD validation with kubebuilder markers. It can be enabled in your configuration.
+
+To enable in `.golangci.yml`:
+```yaml
+linters-settings:
+  custom:
+    kubeapilinter:
+      settings:
+        linters:
+          enable:
+            - numericbounds
+```
+
+See the [package documentation](https://pkg.go.dev/sigs.k8s.io/kube-api-linter/pkg/analysis/numericbounds) for detailed examples and usage.
```

Comment:
We don't have this for other linters so can probably skip unless there is any specific configuration for this rule